webplugin/js/app/km-message-markup-1.0.js

```diff
@@ -109,21 +109,62 @@ Kommunicate.popupChatTemplate = {
             popupWidgetContent &&
             popupWidgetContent.length &&
             popupWidgetContent[0].message;
+        var buttonDetails = popupWidgetContent &&
+            popupWidgetContent.length &&
+            popupWidgetContent[0].buttons;
+        var templateKey =
+            (popupWidgetContent &&
+                popupWidgetContent.length &&
+                popupWidgetContent[0].templateKey) ||
+            KommunicateConstants.CHAT_POPUP_TEMPLATE.HORIZONTAL;
+
+        if(Array.isArray(popupMessageContent)){
+            var len = popupMessageContent.length;
+            if(!len){ popupMessageContent = ""; }
```

Comment:
i guess if .length property is there, it will be of type string only. so no need of this custom empty string conversion?

Reply:
Actually, we can pass an array of greeting messages now.
Out the provided messages any one message is selected randomly.

Note: The old implementation, where we can only pass string, will not be affected.